proof.go

```diff
@@ -248,64 +244,70 @@ func (proof Proof) VerifyNamespace(h hash.Hash, nID namespace.ID, leaves [][]byt
 	return res
 }
 
-// The VerifyLeafHashes function checks whether the given proof is a valid Merkle
-// range proof for the leaves in the leafHashes input. It returns true or false accordingly.
-// If there is an issue during the proof verification e.g., a node does not conform to the namespace hash format, then a proper error is returned to indicate the root cause of the issue.
-// The leafHashes parameter is a list of leaf hashes, where each leaf hash is represented
-// by a byte slice.
-// The size of leafHashes should match the proof range i.e., end-start.
-// If the verifyCompleteness parameter is set to true, the function also checks
-// the completeness of the proof by verifying that there is no leaf in the
-// tree represented by the root parameter that matches the namespace ID nID
-// outside the leafHashes list.
-func (proof Proof) VerifyLeafHashes(nth *NmtHasher, verifyCompleteness bool, nID namespace.ID, leafHashes [][]byte, root []byte) (bool, error) {
+// ValidateProofStructure checks ranges, leaf and node formats, and input compatibility.
```

Comment:
What is input compatibility? This godoc could be clearer. For example

```suggestion
// ValidateProofStructure returns an error if the proof's range is invalid, or the format of leafHash is incorrect, or ...
```

Reply:
unexported